File: core/src/remove/tanks/game/graphics/shader/ShaderProgramPrefab.java
package remove.tanks.game.graphics.shader;

import com.google.common.base.MoreObjects;
import remove.tanks.game.utility.Prefab;

/**
 * @author Mateusz Długosz
 */
public final class ShaderProgramPrefab extends Prefab {
    private final String vertexShaderFilename;
    private final String fragmentShaderFilename;

    public ShaderProgramPrefab(String vertexShaderFilename, String fragmentShaderFilename) {
        this.vertexShaderFilename = vertexShaderFilename;
        this.fragmentShaderFilename = fragmentShaderFilename;
    }

    public String getVertexShaderFilename() {
        return vertexShaderFilename;
    }

    public String getFragmentShaderFilename() {
        return fragmentShaderFilename;
    }

    @Override
    public String toString() {
        return MoreObjects.toStringHelper(this)
                .add("vertexShaderFilename", vertexShaderFilename)
                .add("fragmentShaderFilename", fragmentShaderFilename)
                .toString();
    }
}

File: tools/docs-generator/lib/graphics/shader/shader_program_prefab.py
class ShaderProgramPrefab(object):
    def __init__(self, vertex_shader_filename, fragment_shader_filename):
        self.vertex_shader_filename = str(vertex_shader_filename)
        self.fragment_shader_filename = str(fragment_shader_filename)

    def get_vertex_shader_filename(self):
        return self.vertex_shader_filename

    def get_fragment_shader_filename(self):
        return self.fragment_shader_filename

    def __str__(self):
        return "ShaderProgramPrefab(vertex_shader_filename={}, fragment_shader_filename={})" \
                    .format(self.vertex_shader_filename, self.fragment_shader_filename)
